shared_model/utils/amount_utils.hpp

```diff
@@ -31,19 +33,17 @@ iroha::expected::Result<std::shared_ptr<shared_model::interface::Amount>,
                         std::shared_ptr<std::string>>
 operator+(const shared_model::interface::Amount &a,
           const shared_model::interface::Amount &b) {
-  // check precisions
-  if (a.precision() != b.precision()) {
-    return iroha::expected::makeError(
-        std::make_shared<std::string>("precision mismatch"));
-  }
-  if (a.intValue() + b.intValue() < a.intValue()
-      || a.intValue() + b.intValue() < b.intValue()) {
+  auto max_precision = std::max(a.precision(), b.precision());
+  auto val_a = a.intValue() * (int)std::pow(10, max_precision - a.precision());
+  auto val_b = b.intValue() * (int)std::pow(10, max_precision - b.precision());
```

Comment:
One of these multiplications is not needed since max_precision is either a.precision() or b.precision()